hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/replicas/ReplicasVerify.java

```diff
@@ -216,6 +228,34 @@ void processKey(OzoneClient ozoneClient, String volumeName, String bucketName, S
     }
   }
 
+  /**
+   * Check if the key should be filtered based on replication config.
+   * @param keyInfo the key to check
+   * @return true if the key should be filtered (skipped), false otherwise
+   */
+  private boolean shouldFilterKey(OmKeyInfo keyInfo) {
+    Optional<ReplicationConfig> filterConfig = replication.fromParams(getConf());
+    if (!filterConfig.isPresent()) {
+      // No filter specified, include all keys
+      return false;
+    }
+
+    ReplicationConfig keyReplicationConfig = keyInfo.getReplicationConfig();
+    ReplicationConfig filter = filterConfig.get();
+
+    // Check replication type match
+    if (!keyReplicationConfig.getReplicationType().equals(filter.getReplicationType())) {
+      return true;
+    }
+
+    // Check replication factor/config match
+    if (!keyReplicationConfig.getReplication().equals(filter.getReplication())) {
```

Comment:
These can be combined:
```suggestion
        // Ignore key if replication type and factor/config don't match
    if (!keyReplicationConfig.getReplicationType().equals(filter.getReplicationType())  
    && !keyReplicationConfig.getReplication().equals(filter.getReplication())) {
```